runtime/fastly/builtins/fastly.cpp

```diff
@@ -463,8 +494,59 @@ bool install(api::Engine *engine) {
       // options.getExperimentalHighResolutionTimeMethodsEnabled() ? nowfn : end,
       end};
 
-  return JS_DefineFunctions(ENGINE->cx(), fastly, methods) &&
-         JS_DefineProperties(ENGINE->cx(), fastly, Fastly::properties);
+  return JS_DefineFunctions(engine->cx(), fastly, methods) &&
+         JS_DefineProperties(engine->cx(), fastly, Fastly::properties);
+}
+
+// We currently support five types of body inputs:
+// - byte sequence
+// - buffer source
+// - USV strings
+// - URLSearchParams
+// After the other other options are checked explicitly, all other inputs are
+// encoded to a UTF8 string to be treated as a USV string.
+// TODO: Support the other possible inputs to Body.
+JS::Result<std::tuple<JS::UniqueChars, size_t>> convertBodyInit(JSContext *cx,
+                                                                JS::HandleValue bodyInit) {
```

Comment:
Is there an equivalent of `js-compute-builtins.cpp` in the `fastly` tree? I remember this function being needed in a bunch of other places, so it might be worth moving it in the future.

Reply:
A lot of what was there previously ended up going into StarlingMonkey as separate builtin extensions / in the core. I think in general having helper functions associated with their spec builtins would be nicer than maintaining the generic common when possible.